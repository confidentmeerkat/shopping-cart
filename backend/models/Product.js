const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  info: {
    name: String,
    dimensions: String,
    weight: String,
    displayType: String,
    displaySize: String,
    displayResolution: String,
    os: String,
    cpu: String,
    internalMemory: String,
    ram: String,
    camera: String,
    batery: String,
    color: String,
    price: Number,
    photo: String
  },
  tags: {
    priceRange: String,
    brand: String,
    color: String,
    os: String,
    internalMemory: String,
    ram: String,
    displaySize: String,
    displayResolution: String,
    camera: String,
    cpu: String
  }
});

module.exports = mongoose.model('Product', productSchema);