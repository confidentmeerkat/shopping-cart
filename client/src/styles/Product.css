.product {
  display: flex;
  width: 100%;
  height: 200px;
  max-height: 200px;
  margin: 10px 0;
  box-shadow: 0 0 7px #d6d6d6;
}

.product img {
  max-width: 190px;
  max-height: 180px;
  margin: auto;
}

.content {
  display: flex;
  width: 75%;
  height: 190px;
  max-height: 190px;
  margin: auto;
}

.content-left {
  width: 80%;
}

.content-left h3 {
  margin: 10px 0 25px 0;
}

.content-left div {
  margin: 1.5px 0;
}

.content-right {
  width: 20%;
}

.content-right p {
  margin: 50px 0 0 0;
}

.content-right h2 {
  margin: 0 0 40px 0;
  font-size: 30px;
  color: #64DD17;
}