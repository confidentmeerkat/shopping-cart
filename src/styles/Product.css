.product {
  display: flex;
  width: 98%;
  height: 200px;
  max-height: 200px;
  margin: 10px 0 10px 20px;
  box-shadow: 0 0 7px #d6d6d6;
}

.product img {
  max-width: 190px;
  max-height: 180px;
  margin: auto;
}

.product .content {
  width: 75%;
  height: 190px;
  max-height: 190px;
  margin: auto;
}

.product .content h3 {
  margin: 10px 0 25px 0;
}

.product .content div {
  margin: 1.5px 0;
}