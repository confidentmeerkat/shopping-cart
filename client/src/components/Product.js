import React from 'react';
import { Link } from 'react-router-dom';
import RaisedButton from 'material-ui/RaisedButton';
import NavigateNext from 'material-ui/svg-icons/image/navigate-next';
import '../styles/Product.css';

const Product = ({ item }) => (
  <div className="product">
    <img src={item.info.photo} />
    <div className="content">
      <div className="content-left">
        <h3>{item.info.name}</h3>
        <div><b>Display size: </b><span>{item.info.displaySize}</span></div>
        <div><b>Display resolution: </b><span>{item.info.displayResolution}</span></div>
        <div><b>CPU: </b><span>{item.info.cpu}</span></div>
        <div><b>Internal memory: </b><span>{item.info.internalMemory}</span></div>
        <div><b>RAM: </b><span>{item.info.ram}</span></div>
        <div><b>Camera: </b><span>{item.info.camera}</span></div>
      </div>
      <div className="content-right">
        <h2>${item.info.price}.00</h2>
        <RaisedButton
          containerElement={<Link to={`/product/${item.id}`} />}
          className="btn"
          label="See more"
          labelPosition="before"
          primary={true}
          icon={<NavigateNext />}
        />
      </div>
    </div>
  </div>
);

export default Product;