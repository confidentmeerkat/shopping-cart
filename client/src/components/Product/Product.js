import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import RaisedButton from 'material-ui/RaisedButton';
import NavigateNext from 'material-ui/svg-icons/image/navigate-next';
import '@styles/Product.css';

const Product = (props) => {
  const { item } = props;
  const { info: {
    photo,
    name,
    displaySize,
    displayResolution,
    cpu,
    internalMemory,
    ram,
    camera,
    price
  }} = item;

  return (
    <div className="product">
      <div className="content">
        <img src={photo} />
        <div className="content-left">
          <h3>{name}</h3>
          <div className="content-info">
            <div><b>Display size: </b><span>{displaySize}</span></div>
            <div><b>Display resolution: </b><span>{displayResolution}</span></div>
            <div><b>CPU: </b><span>{cpu}</span></div>
            <div><b>Internal memory: </b><span>{internalMemory}</span></div>
            <div><b>RAM: </b><span>{ram}</span></div>
            <div><b>Camera: </b><span>{camera.length < 50 ? camera : camera.slice(0, 50) + '...'}</span></div>
          </div>
        </div>
        <div className="content-right">
          <div className="content-info">
            <p><b>Price:</b></p>
            <h2>{numeral(price).format('$0,0.00')}</h2>
          </div>
          <RaisedButton
            containerElement={<Link to={`/product/${item._id}`} />}
            className="btn"
            label="See more"
            labelPosition="before"
            primary={true}
            icon={<NavigateNext />}
          />
        </div>
      </div>
    </div>
  )
};

export default Product;
