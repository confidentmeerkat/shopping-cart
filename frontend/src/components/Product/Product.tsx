import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import RaisedButton from 'material-ui/RaisedButton';
import NavigateNext from 'material-ui/svg-icons/image/navigate-next';
import { ICatalogProduct } from '@typings/state/index'
import '@styles/Product.css';

interface Props {
  key: string;
  item: ICatalogProduct;
}

const Product = ({ item: {info, _id} }: Props) => {
  const {
    photo,
    name,
    displaySize,
    displayResolution,
    cpu,
    internalMemory,
    ram,
    camera,
    price
  } = info;

  return (
    <div className="product">
      <div className="content">
        <img src={photo} />
        <div className="content-left">
          <h3>{name}</h3>
          <div className="content-info">
            <div><b>Display size: </b><span>{displaySize}</span></div>
            <div><b>Display resolution: </b><span>{displayResolution}</span></div>
            <div><b>CPU: </b><span>{cpu}</span></div>
            <div><b>Internal memory: </b><span>{internalMemory}</span></div>
            <div><b>RAM: </b><span>{ram}</span></div>
            <div><b>Camera: </b><span>{camera.length < 50 ? camera : camera.slice(0, 50) + '...'}</span></div>
          </div>
        </div>
        <div className="content-right">
          <div className="content-info">
            <p><b>Price:</b></p>
            <h2>{numeral(price).format('$0,0.00')}</h2>
          </div>
          <RaisedButton
            containerElement={<Link to={`/product/${_id}`} />}
            className="btn"
            label="See more"
            labelPosition="before"
            primary={true}
            icon={<NavigateNext />}
          />
        </div>
      </div>
    </div>
  )
};

export default Product;
