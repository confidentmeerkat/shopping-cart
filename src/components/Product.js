import React from 'react';
import '../styles/Product.css';

const Product = () => (
  <div className="product">
    <img src="./img/a3.JPG" />
    <div className="content">
      <h3>SAMSUNG A3 2015</h3>
      <div><b>Display size: </b><span>15 inches</span></div>
      <div><b>Display resolution: </b><span>1920x1440</span></div>
      <div><b>CPU: </b><span>1920x1440</span></div>
      <div><b>Internal memory: </b><span>1920x1440</span></div>
      <div><b>RAM: </b><span>1920x1440</span></div>
      <div><b>Camera: </b><span>1920x1440</span></div>
    </div>
  </div>
);

export default Product;