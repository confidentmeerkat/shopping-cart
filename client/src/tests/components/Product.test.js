import React from 'react';
import { shallow } from 'enzyme';
import Product from '../../components/Product';
import item from '../__mocks__/item';

describe('<Product />', () => {
  it('renders Product correctly', () => {
    const wrapper = shallow(<Product item={item} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.content-info').at(0).children()).toHaveLength(6);
    expect(wrapper.find('img').prop('src')).toBe('/img/samsung_galaxy_s8.jpg');
    expect(wrapper.find('.content-info').at(0).childAt(5).text()).toBe('Camera: 12 MP (f/1.7, 26mm, 1/2.5\", 1.4 Âµm, Dual Pixel PDA...');
  });
});