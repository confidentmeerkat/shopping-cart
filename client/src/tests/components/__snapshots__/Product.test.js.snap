// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> renders Product correctly 1`] = `
<div
  className="product"
>
  <div
    className="content"
  >
    <img
      src="/img/samsung_galaxy_s8.jpg"
    />
    <div
      className="content-left"
    >
      <h3>
        Samsung Galaxy S8
      </h3>
      <div
        className="content-info"
      >
        <div>
          <b>
            Display size: 
          </b>
          <span>
            5.8"
          </span>
        </div>
        <div>
          <b>
            Display resolution: 
          </b>
          <span>
            1440 x 2960 pixels
          </span>
        </div>
        <div>
          <b>
            CPU: 
          </b>
          <span>
            Octa-core (4x2.3 GHz & 4x1.7 GHz) - EMEA
          </span>
        </div>
        <div>
          <b>
            Internal memory: 
          </b>
          <span>
            64 GB
          </span>
        </div>
        <div>
          <b>
            RAM: 
          </b>
          <span>
            4 GB
          </span>
        </div>
        <div>
          <b>
            Camera: 
          </b>
          <span>
            12 MP (f/1.7, 26mm, 1/2.5", 1.4 Âµm, Dual Pixel PDA...
          </span>
        </div>
      </div>
    </div>
    <div
      className="content-right"
    >
      <div
        className="content-info"
      >
        <p>
          <b>
            Price:
          </b>
        </p>
        <h2>
          $720.00
        </h2>
      </div>
      <RaisedButton
        className="btn"
        containerElement={
          <Link
            replace={false}
            to="/product/undefined"
          />
        }
        disabled={false}
        fullWidth={false}
        icon={<ImageNavigateNext />}
        label="See more"
        labelPosition="before"
        primary={true}
        secondary={false}
      />
    </div>
  </div>
</div>
`;
